tests/core/creators/detached.js

```diff
@@ -0,0 +1,150 @@
+			var editorElement = CKEDITOR.document.getById( 'editor6' );
+
+			var editor = CKEDITOR.replace( editorElement, {
+				delayIfDetached: false
+			} );
+
+			assert.isNotNull( editor );
+		},
+
+		'test editor without config is created immediately on not detached element': function() {
+			var editorElement = CKEDITOR.document.getById( 'editor7' );
+
+			var editor = CKEDITOR.replace( editorElement );
+
+			assert.isNotNull( editor );
+		},
+
+		'test delay editor creation until target element attach to DOM': function() {
+			var editorElement = CKEDITOR.document.getById( 'editor2' ),
+				editorParent = editorElement.getParent();
+
+			editorElement.remove();
+
+			CKEDITOR.replace( editorElement, {
+				delayIfDetached: true,
+				on: {
+					instanceReady: function() {
+						resume( function() {
+							assert.pass();
+						} );
+					}
+				}
+			} );
+
+			CKEDITOR.tools.setTimeout( function() {
+				editorParent.append( editorElement );
+			}, 250 );
+
+			wait();
+		},
+
+		'test editor creation from provided callback': function() {
+			var editorElement = CKEDITOR.document.getById( 'editor3' ),
+				editorParent = editorElement.getParent(),
+				editorCreationCallback;
+
+			editorElement.remove();
+
+			CKEDITOR.replace( editorElement, {
+				delayIfDetached: true,
+				delayIfDetached_callback: registerCallback,
+				on: {
+					instanceReady: function() {
+						resume( function() {
+							assert.pass();
+						} );
+					}
+				}
+			} );
+
+			function registerCallback( editorCreationFunc ) {
+				editorCreationCallback = editorCreationFunc;
+			}
+
+			CKEDITOR.tools.setTimeout( function() {
+				editorParent.append( editorElement );
+				editorCreationCallback();
+			}, 250 );
+
+			wait();
+		},
+
+		'test editor delay creation invokes CKEDITOR.warn': function() {
+			var stubbedWarn = sinon.stub( CKEDITOR, 'warn' ),
+				editorElement = CKEDITOR.document.getById( 'editor4' ),
+				editorParent = editorElement.getParent();
+
+			editorElement.remove();
+
+			CKEDITOR.replace( editorElement, {
+				delayIfDetached: true,
+				on: {
+					instanceReady: function() {
+						resume( function() {
+							assert.isTrue( stubbedWarn.calledTwice );
+							stubbedWarn.restore();
+						} );
+					}
+				}
+			} );
+
+			CKEDITOR.tools.setTimeout( function() {
+				editorParent.append( editorElement );
+			}, 250 );
+
+			wait();
+		},
+
+		'test editor interval attempts to create if target element is detached': function() {
+			var editorElement = CKEDITOR.document.getById( 'editor8' ),
+				spyIsDetached = sinon.spy( editorElement, 'isDetached' );
+
+			editorElement.remove();
+
+			CKEDITOR.replace( editorElement, {
+				delayIfDetached: true,
+				delayIfDetached_interval: 50
+			} );
+
+			CKEDITOR.tools.setTimeout( function() {
+				resume( function() {
+					assert.isTrue( spyIsDetached.callCount > 5 );
+					spyIsDetached.restore();
+				} );
+			}, 3000 );
```

Comment:
This is a clever approach and I like it 👍🏻  I only propose to reduce the timeout to not extend the test longer than we need it - just 2 calls should make us comfortable, so `1000` would be enough for that.

Reply:
I go even with 200. With `50`ms interval, it is enough to make at least 2 calls. Also, there is also one `isDetached` call in `replace` method. Since it's a unit test - those `800`ms might be valuable in a bigger test run.  
  
Also, I have to fix one bleeding `setInterval` in `test delay editor creation if target element is detached`. So after the test verifies the result - I attach the editor again. If we don't do that - this spy is counting still 'ticking' interval from the previous test.